File: freecodecamp/daily-code-challenge/2026-04-05.js
function getRotation(n) {
    let s = String(n);
    const k = s.length;

    for (let i = 0; i < k; i++) {
        let num = Number(s);

        if (num % k === 0) {
            return i;
        }

        // rotate: move first char to end
        s = s.slice(1) + s[0];
    }

    return "none";
}

File: freecodecamp/daily-code-challenge/2026-04-05.py
def get_rotation(n):
    s = str(n)
    k = len(s)

    for i in range(k):
        num = int(s)

        if num % k == 0:
            return i

        # rotate: move first digit to end
        s = s[1:] + s[0]

    return "none"